bitcoinlib/services/groestlcoind.py

```diff
@@ -0,0 +1,271 @@
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from datetime import datetime
+from bitcoinlib.main import *
+from bitcoinlib.services.authproxy import AuthServiceProxy
+from bitcoinlib.services.baseclient import BaseClient, ClientError
+from bitcoinlib.transactions import Transaction
+from bitcoinlib.encoding import to_hexstring
+from bitcoinlib.networks import Network
+
+
+PROVIDERNAME = 'groestlcoind'
+
+_logger = logging.getLogger(__name__)
+
+
+class ConfigError(Exception):
+    def __init__(self, msg=''):
+        self.msg = msg
+        _logger.info(msg)
+
+    def __str__(self):
+        return self.msg
+
+try:
+    import configparser
+except ImportError:
+    import ConfigParser as configparser
+
+
+def _read_from_config(configparser, section, value, fallback=None):
+    try:
+        return configparser.get(section, value)
+    except Exception:
+        return fallback
+
+
+class GroestlcoindClient(BaseClient):
+    """
+    Class to interact with groestlcoind, the Bitcoin deamon
+    """
+
+    @staticmethod
+    def from_config(configfile=None, network='bitcoin'):
+        """
+        Read settings from groestlcoind config file
+
+        :param configfile: Path to config file. Leave empty to look in default places
+        :type: str
+        :param network: Bitcoin mainnet or testnet. Default is bitcoin mainnet
+        :type: str
+
+        :return GroestlcoindClient:
+        """
+        try:
+            config = configparser.ConfigParser(strict=False)
+        except TypeError:
+            config = configparser.ConfigParser()
+        config_fn = 'bitcoin.conf'
+        if isinstance(network, Network):
+            network = network.name
+        if network == 'testnet':
+            config_fn = 'bitcoin-testnet.conf'
+
+        cfn = None
+        if not configfile:
+            config_locations = ['~/.bitcoinlib', '~/.bitcoin', '~/Application Data/Bitcoin',
+                                '~/Library/Application Support/Bitcoin']
+            for location in config_locations:
+                cfn = Path(location, config_fn).expanduser()
+                if cfn.exists():
+                    break
+        else:
+            cfn = Path(BCL_DATA_DIR, 'config', configfile)
+        if not cfn or not cfn.is_file():
+            raise ConfigError("Config file %s not found. Please install bitcoin client and specify a path to config "
+                              "file if path is not default. Or place a config file in .bitcoinlib/bitcoin.conf to "
+                              "reference to an external server." % cfn)
```

Comment:
raise ConfigError("Config file %s not found. Please install groestlcoin client and specify a path to config "
                              "file if path is not default. Or place a config file in .bitcoinlib/groestlcoin.conf to "
                              "reference to an external server." % cfn)